### PR TITLE
[Feature] Add support for stop limit orders

### PR DESCRIPTION
**Pull request by BraveSirAndrew**

### CODE REVIEW

File: GDAXSharp/Services/Orders/Models/Order.cs [hunk 1]
```diff
@@ -17,6 +17,13 @@ public class Order
 
         public decimal Price { get; set; }
 
+	    [JsonProperty("stop")]
+	    [JsonConverter(typeof(StringEnumConverter))]
+		public StopType StopType { get; set; }
+
+		[JsonProperty("stop_price")]
```
Comment (dougdellolio) [inline GDAXSharp/Services/Orders/Models/Order.cs]: please fix the spacing here to line up with other properties

File: GDAXSharp/Services/Orders/OrdersService.cs [hunk 2]
```diff
@@ -89,6 +89,31 @@ public class OrdersService : AbstractService
             return await PlaceOrderAsync(order);
         }
 
+	    public async Task<OrderResponse> PlaceStopLimitOrderAsync(
```
Comment (dougdellolio) [inline GDAXSharp/Services/Orders/OrdersService.cs]: spacing here looks weird. should be consistent with the others. something like this:

````
        public async Task<OrderResponse> PlaceStopLimitOrderAsync(
            OrderSide side,
            ProductType productId,
            decimal size,
            decimal stopPrice,
            decimal limitPrice,
            bool postOnly = true,
            Guid? clientOid = null)
        {
            var order = new Order
            {
                Side = side,
                ProductId = productId,
                OrderType = OrderType.Limit,
                Price = limitPrice,
                Size = size,
                StopType = side == OrderSide.Buy
                    ? StopType.Entry
                    : StopType.Loss,
                StopPrice = stopPrice,
                ClientOid = clientOid,
                PostOnly = postOnly
            };

            return await PlaceOrderAsync(order);
        }
````

File: GDAXSharp/Services/Orders/Types/StopType.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+namespace GDAXSharp.Services.Orders.Types
+{
+	public enum StopType
+	{
+		[EnumMember(Value = "loss")]
+		Loss,
+		[EnumMember(Value = "entry")]
+		Entry
+	}
+}
```
Comment (dougdellolio) [inline GDAXSharp/Services/Orders/Types/StopType.cs]: please add a nl

**Comment by BraveSirAndrew** [comment]
Yep, tested on sandbox and live servers. Seems to work fine. It's sort of indirectly referenced in their docs, but cobbled it together mainly from other sources on Github.

**Pull request merged**